lib/customisation.nix

```diff
@@ -66,28 +66,31 @@ rec {
   */
   makeOverridable = f: origArgs:
     let
-      ff = f origArgs;
+      result = f origArgs;
+
       # Creates a functor with the same arguments as f
       copyArgs = g: lib.setFunctionArgs g (lib.functionArgs f);
+      # Changes the original arguments with (potentially a function that returns) a set of new attributes
       overrideWith = newArgs: origArgs // (if lib.isFunction newArgs then newArgs origArgs else newArgs);
 
       # Re-call the function but with different arguments
       overrideArgs = copyArgs (newArgs: makeOverridable f (overrideWith newArgs));
       # Change the result of the function call by applying g to it
       overrideResult = g: makeOverridable (copyArgs (args: g (f args))) origArgs;
     in
-      if builtins.isAttrs ff then (ff // {
-        override = overrideArgs;
-        overrideDerivation = fdrv: overrideResult (x: overrideDerivation x fdrv);
-        ${if ff ? overrideAttrs then "overrideAttrs" else null} = fdrv:
-          overrideResult (x: x.overrideAttrs fdrv);
-      })
-      else if lib.isFunction ff then
-        # Transform ff into a functor while propagating its arguments
-        lib.setFunctionArgs ff (lib.functionArgs ff) // {
+      if builtins.isAttrs result then
+        result // {
+          override = overrideArgs;
+          overrideDerivation = fdrv: overrideResult (x: overrideDerivation x fdrv);
+          ${if result ? overrideAttrs then "overrideAttrs" else null} = fdrv:
+            overrideResult (x: x.overrideAttrs fdrv);
+        }
+      else if lib.isFunction result then
+        # Transform the result into a functor while propagating its arguments
+        lib.setFunctionArgs result (lib.functionArgs result) // {
```

Comment:
You defined `copyArgs` in an earlier commit, so why not use it here as well?

Reply:
It's not the same, `copyArgs` makes a function have the args of `f`, while this is `result`